### PR TITLE
Reformatting travis config file

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
cleaned up travis.yml file to look nicer

### Implementation details
Reformatted to use matrix construct.
Updated go version from 1.9 to 1.11, and fixed new go vet failures from it:
- composite literals using unkeyed field
- copying lock value 

### Testing

New tests cover the changes: N/A no agent code change made

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,17 @@
 language: go
+go_import_path: github.com/aws/amazon-ecs-agent
 sudo: false
 go:
   - 1.9
```
Comment (petderek) [inline .travis.yml]: any reason we can't bump this to 1.11?

File: .travis.yml [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
+  include:
+    - os: linux
+      script:
+        - make get-deps
+        - make test
+        - make static-check
+        - make xplatform-build
+    - os: windows
+      script:
+        - go test -race -tags unit -timeout 40s ./agent/...
```
Comment (petderek) [inline .travis.yml]: add a '-v' in there (i missed it the first time around...)

File: .travis.yml (hunk 1 above)
Comment (yunhee-l) [reply .travis.yml]: fixed

File: .travis.yml (hunk 2 above)
Comment (yunhee-l) [reply .travis.yml]: fixed

**Comment by petderek** [comment]
It looks like the newer version of `go vet` is getting mad. Fix the error and then 🐑 ?

**Pull request merged**